source/custom/nixnetsocket_service.custom.cpp

```diff
@@ -2,26 +2,26 @@
 
 namespace nixnetsocket_grpc {
 
-::grpc::Status NiXnetSocketService::ConvertApiErrorStatusForNxSOCKET(google::protobuf::int32 status, nxSOCKET socket)
+::grpc::Status NiXnetSocketService::ConvertApiErrorStatusForNxSOCKET(::grpc::ServerContext* context, google::protobuf::int32 status, nxSOCKET socket)
 {
-    // This implementation assumes this method is always called on the same thread where the error occurred.
-    std::string description(nidevice_grpc::kMaxGrpcErrorDescriptionSize, '\0');
-    library_->GetLastErrorStr(&description[0], nidevice_grpc::kMaxGrpcErrorDescriptionSize);
-    // XNET sockets have special behavior that returns both the API error code and the sockets error number.
-    int errorNumber = library_->GetLastErrorNum();
-    nlohmann::json jsonError{{"errorNumber", errorNumber}};
-    return nidevice_grpc::ApiErrorAndDescriptionToStatus(status, description, jsonError);
+  // This implementation assumes this method is always called on the same thread where the error occurred.
+  std::string description(nidevice_grpc::kMaxGrpcErrorDescriptionSize, '\0');
+  library_->GetLastErrorStr(&description[0], nidevice_grpc::kMaxGrpcErrorDescriptionSize);
+  // XNET sockets have special behavior that returns both the API error code and the sockets error number.
+  int errorNumber = library_->GetLastErrorNum();
+  context->AddTrailingMetadata("socket-error", std::to_string(errorNumber));
```

Comment:
Let's prefix our custom metadata with `ni-` to ensure that it never conflicts with [standard HTTP headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers).

(This is more important for headers than for trailers, but I think it's a good practice to follow in both cases.)